modules/administration-guide/attachments/restricted-environment/prepare-restricted-environment.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+#
+# Copyright (c) 2022 Red Hat, Inc.
+# This program and the accompanying materials are made
+# available under the terms of the Eclipse Public License 2.0
+# which is available at https://www.eclipse.org/legal/epl-2.0/
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+
+# OpenShift cluster version
+export ocp_ver="${ocp_ver:-4.10}"
+
+# Destination registry
+export my_registry="${my_registry:-default-route-openshift-image-registry.apps-crc.testing}"
+export my_catalog="${my_catalog:-airgap-che}"
+export my_operator_index="$my_registry/$my_catalog/my-operator-index:v${ocp_ver}"
+
+# Operators
+export prod_operator="${prod_operator:-eclipse-che}"
+export prod_operator_bundle="${prod_operator_bundle:-quay.io/openshift-community-operators/eclipse-che:v7.47.0}"
+export devworkspace_operator_bundle="${devworkspace_operator_bundle:-quay.io/devfile/devworkspace-operator-index:release}"
+
+# Create local directories
+mkdir -p "${my_catalog}/${prod_operator}"
+mkdir -p "${my_catalog}/devworkspace"
+
+# Get metadata for product and devworkspace operators
+opm render "${prod_operator_bundle}" > "${my_catalog}/${prod_operator}/render.json"
+opm render "${devworkspace_operator_bundle}" > "${my_catalog}/devworkspace/render.json"
```

Comment:
Rendering only bundles is insufficient. Building a file-based catalog requires also defining a `olm.channel` and `olm.package`.

If you would like to include the full DevWorkspace operator release, you should be rendering `quay.io/devfile/devworkspace-operator-index:release`, which includes all bundles and the required channels/packages.

Reply:
And for Che?

Reply:
For Che, you probably have to render + filter the community catalog (I don't know the image offhand). This is more complicated -- the simplest solution would be
```bash
# The render below takes over a minute
opm render quay.io/operatorhubio/catalog:latest > render.json
jq 'select(.name == "eclipse-che" or .package == "eclipse-che")' render.json > "${my_catalog}/${prod_operator}/render.json"
rm -f render.json 
```
but this would need adapting if you want to automatically downstream this script as well.

Reply:
Implemented in the last iteration.